src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/SimdVectorExtensions.cs

```diff
@@ -0,0 +1,171 @@
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The array to which <paramref name="vector" /> is copied.</param>
+        /// <param name="startIndex">The starting index of <paramref name="destination" /> which <paramref name="vector" /> will be copied to.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="ISimdVector{TVector, T}.Count" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="startIndex" /> is negative or greater than the length of <paramref name="destination" />.</exception>
+        /// <exception cref="NullReferenceException"><paramref name="destination" /> is <c>null</c>.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        public static void CopyTo<TVector, T>(this TVector vector, T[] destination, int startIndex)
+            where TVector : ISimdVector<TVector, T>
+        {
+            TVector.CopyTo(vector, destination, startIndex);
+        }
+
+        /// <summary>Copies a vector to a given span.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector to be copied.</param>
+        /// <param name="destination">The span to which the <paramref name="vector" /> is copied.</param>
+        /// <exception cref="ArgumentException">The length of <paramref name="destination" /> is less than <see cref="ISimdVector{TVector, T}.Count" />.</exception>
+        /// <exception cref="NotSupportedException">The type of the elements in the vector (<typeparamref name="T" />) is not supported.</exception>
+        public static void CopyTo<TVector, T>(this TVector vector, Span<T> destination)
+            where TVector : ISimdVector<TVector, T>
+        {
+            TVector.CopyTo(vector, destination);
+        }
+
+        /// <summary>Gets the element at the specified index.</summary>
+        /// <typeparam name="TVector">The type of the vector.</typeparam>
+        /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+        /// <param name="vector">The vector to get the element from.</param>
+        /// <param name="index">The index of the element to get.</param>
+        /// <returns>The value of the element at <paramref name="index" />.</returns>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="index" /> was less than zero or greater than the number of elements.</exception>
+        /// <exception cref="NotSupportedException">The type of <paramref name="vector" /> (<typeparamref name="T" />) is not supported.</exception>
+        public static T GetElement<TVector, T>(this TVector vector, int index)
```

Comment:
The indexer can't be part of the interface?

Reply:
We have a pattern of making various APIs extensions rather than instance methods for efficiency purposes (instance methods take `TVector` by ref, which can make the IL and other code to be larger and more complex. That in turn causes the JIT to do more work and can mess up some optimizations.

Having the extension then allows you to access the API like an instance method, without forcing the operand be address taken, without forcing the introduction of additional locals, and so on.